### ISSUE TITLE
[Gemma3][Text] Softcapping in SDPA

### ISSUE THREAD

**Issue by ppetrovicTT** [tenstorrent/tt-metal#22807]
Gemma3 adds softcapping within SDPA block.

We have an SDPA operator in C++, so we should add this possibility there:
https://github.com/tenstorrent/tt-metal/tree/main/ttnn/cpp/ttnn/operations/transformer/sdpa_decode

Example before softmax from Gemma3 Pytorch:
https://github.com/google/gemma_pytorch/blob/main/gemma/model.py#L324
```
if self.attn_logit_softcapping is not None:
            scores = scores / self.attn_logit_softcapping
            scores = torch.tanh(scores)
            scores = scores * self.attn_logit_softcapping

        scores = scores + mask
        scores = F.softmax(scores.float(), dim=-1).type_as(q)
```

**Comment by ppetrovicTT** [comment]
The first implementation is in [22807-softcapping-sdpa](https://github.com/tenstorrent/tt-metal/tree/ppetrovic/22807-softcapping-sdpa) branch.
But, I'm seeing low accuracy due to tanh implementation in llk.

By further examination, it seems that attn_logit_softcapping parameter is used only in Gemma2, and not Gemma3.

Once everything else is aligned, I'll raise the ticket to LLK team for the tanh precision, if needed at all.

**Comment by ricozhu-TT** [comment]
@ppetrovicTT I can pick this up for Grok 2.5 bring-up. I have a (currently hard-coded for Grok but will generalize soon) working implementation on the [grok25 branch](https://github.com/tenstorrent/tt-metal/tree/grok25) re-using your code. Since we defer the scaling by head_sim ** -0.5 to the exp call via `sub_exp`, the mathematically equivalent formulation would be to defer the softcap scaling to sub_exp, and then in the softcap template call scaling by (head_dim ** -0.5) * (softcap_scale ** -1), then apply tanh as before.

**Comment by ppetrovicTT** [comment]
We put this on hold, as it turned out it's a Gemma2 feature, not Gemma3.
If you can reuse anything, please feel free! We won't be pushing it for Gemma3.
Thank you!

**Comment by ricozhu-TT** [comment]
No worries, looks like this is really only needed for Gemma 2 and Grok 2.5 anyways.

**Comment by ppetrovicTT** [comment]
This feature is required by Gemma2, and not used in Gemma3.
We don't plan to support Gemma2, only Gemma3, so I'm closing this as not needed.
